sklearn/metrics/pairwise.py

```diff
@@ -1427,6 +1655,19 @@ def _check_chunk_size(reduced, chunk_size):
 def _precompute_metric_params(X, Y, metric=None, **kwds):
     """Precompute data-derived metric parameters if not provided
     """
+    if metric == 'gower':
+        categorical_features = None
+        if 'categorical_features' in kwds:
+            categorical_features = kwds['categorical_features']
+
+        num_mask = ~ _detect_categorical_features(X, categorical_features)
+
+        scale = None
+        if 'scale' in kwds:
+            scale = kwds['scale']
+        scale, _, _ = _precompute_gower_params(X, Y, scale, num_mask)
+
+        return {'scale': scale}
```

Comment:
shouldn't we also return the determined `categorical_features` if they had been passed in as `None`?